bundles/org.openhab.io.homekit/src/main/java/org/openhab/io/homekit/internal/accessories/HomekitCharacteristicFactory.java

```diff
@@ -123,28 +127,29 @@
     /**
      * create optional HomeKit characteristic
      *
-     * @param type type of characteristic
      * @param item corresponding OH item
      * @param updater update to keep OH item and HomeKit characteristic in sync
      * @return HomeKit characteristic
      */
-    public static Characteristic createCharacteristic(HomekitCharacteristicType type, GenericItem item,
-            HomekitAccessoryUpdater updater) throws HomekitException {
-        logger.trace("createCharacteristic, type {} item {}", type, item);
-        if (optional.containsKey(type)) {
-            return optional.get(type).apply(item, updater);
+    public static Characteristic createCharacteristic(final HomekitTaggedItem item, HomekitAccessoryUpdater updater)
+            throws HomekitException {
+        logger.trace("createCharacteristic, type {} item {}", item.getCharacteristicType(), item);
```

Comment:
Please cache `item.getCharacteristicType()` in a local variable for reuse.

Reply:
good one. need to pay more attention to this.